examples/seismic/skew_self_adjoint/example_iso.py

```diff
@@ -1,23 +1,93 @@
 import numpy as np
+from argparse import ArgumentParser
 
-from examples.seismic import RickerSource
-from examples.seismic.skew_self_adjoint import (default_setup_iso,
+from devito.logger import info
+from devito import configuration, smooth, Function
+
+from examples.seismic import setup_geometry, Model
+from examples.seismic.skew_self_adjoint import (setup_w_over_q,
                                                 SsaIsoAcousticWaveSolver)
 
-shape = (221, 211, 201)
-dtype = np.float32
-npad = 20
-qmin = 0.1
-qmax = 1000.0
-tmax = 500.0
-fpeak = 0.010
-omega = 2.0 * np.pi * fpeak
 
-b, v, time_axis, src_coords, rec_coords = default_setup_iso(npad, shape, dtype, tmax=tmax)
+def acousticssa_setup(shape=(50, 50, 50), spacing=(10.0, 10.0, 10.0),
+                      tn=500., space_order=8, nbl=10, **kwargs):
+    # SSA parameters
+    qmin = 0.1
+    qmax = 1000.0
+    tmax = 500.0
+    fpeak = 0.010
+    omega = 2.0 * np.pi * fpeak
+    vp = 1.5*np.ones(shape)
+    b = 1.0*np.ones(shape)
+
+    init_damp = lambda func, nbl: setup_w_over_q(func, omega, qmin, qmax, nbl, sigma=0)
+    o = tuple([0]*len(shape))
+    spacing = spacing[:len(shape)]
+    model = Model(origin=o, shape=shape, vp=vp, b=b, spacing=spacing, nbl=nbl,
+                  space_order=space_order, bcs=init_damp,
+                  dtype=kwargs.pop('dtype', np.float32), **kwargs)
+    # Source and receiver geometries
+    geometry = setup_geometry(model, tmax)
+
+    # Create solver object to provide relevant operators
+    solver = SsaIsoAcousticWaveSolver(model, geometry,
+                                      space_order=space_order, **kwargs)
+    return solver
+
+
+def run(shape=(50, 50, 50), spacing=(10.0, 10.0, 10.0), tn=1000.0,
+        space_order=4, nbl=40, full_run=False, autotune=False, **kwargs):
+
+    solver = acousticssa_setup(shape=shape, spacing=spacing, nbl=nbl, tn=tn,
```

Comment:
acoustic_ssa_setup ?

btw, sooner or later we'll have to replace somewhat `acoustic` with `iso-acoustic` for clarity right?

Reply:
yes, but for another time ;)

Reply:
agreed